pyquil/wavefunction.py

```diff
@@ -147,21 +148,34 @@ def pretty_print_probabilities(self, decimal_digits=2):
                 outcome_dict[outcome] = prob
         return outcome_dict
 
-    def pretty_print(self, decimal_digits=2):
+    def pretty_print(self, decimal_digits=2, ancillae: Optional[Sequence[int]] = None):
         """
         Returns a string repr of the wavefunction, ignoring all outcomes with approximately zero
         amplitude (up to a certain number of decimal digits) and rounding the amplitudes to
         decimal_digits.
 
         :param int decimal_digits: The number of digits to truncate to.
+        :param ancillae: The list of indices for ancillary qubits to omit.
         :return: A dict with outcomes as keys and complex amplitudes as values.
         :rtype: str
         """
+
+        # Initialize bitmask as 1's complement of ancilla spec
+        if ancillae is None:
+            ancillae = []
+        mask = 0
+        for ancilla in ancillae:
+            mask |= 1 << ancilla
+
         outcome_dict = {}
         qubit_num = len(self)
         pp_string = ""
         for index, amplitude in enumerate(self.amplitudes):
+            if index & mask != 0 and not np.isclose(amplitude, 0):
```

Comment:
I won't push hard, but I kinda want this to be configurable. It's also possible that ancillae begin in a non-zero state and should end in the same state.

Reply:
@karalekas brought up the same thing [here](https://github.com/rigetti/pyquil/pull/1083#discussion_r359136257). Will fix now

Reply:
That is, I'll remove the need for the ancillae amplitudes to be zero. What did you want to be configurable?

Reply:
Whether the zero-amplitudes thing is enforced.

Reply:
I just removed it entirely. Your call: are there enough instances when you'd want to be able to enforce this? If so, I'll leave it in and add configurability

Reply:
I'm OK with it not existing at all in this function. It's definitely an important restriction, but not one that I would expect to see solely enforced during pretty printing.